firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/SharedPreferencesUtil.java

```diff
@@ -209,11 +209,28 @@ public synchronized void clearModelDetails(@NonNull String modelName, boolean cl
         if (extractModel != null) {
           customModels.add(extractModel);
         }
+      } else {
+        matcher = Pattern.compile(DOWNLOADING_MODEL_ID_PATTERN).matcher(key);
+        if (matcher.find()) {
+          String modelName = matcher.group(matcher.groupCount());
+          CustomModel extractModel = isDownloadCompleted(modelName);
+          if (extractModel != null) {
+            customModels.add(extractModel);
+          }
+        }
       }
     }
     return customModels;
   }
 
+  synchronized CustomModel isDownloadCompleted(String modelName) {
```

Comment:
the `is` prefix in the name suggests a boolean response. What about using `maybe`, like `maybeGetUpdatedModel`? It's a bit more verbose.

Reply:
Done